### ISSUE TITLE
Blog:  Self-host static-site blog at operationcode.org/blog

### ISSUE THREAD

**Issue by jjhampton** [OperationCode/operationcode#624]
As a VISITOR, 
When I visit https://www.operationcode.org/blog, 
I can read Operation Code's self-hosted blog.

Additional Information:
* Prefer to use https://www.operationcode.org/blog URL instead of using a subdomain, to optimize SEO - see discussion in #576

Potential tools: 
* Jekyll - Easy to build, use, and maintain.  Members of the OC community can help to build from scratch.
* ~Federalist (open-source interface for publishing static websites) from 18F.  May be able to use with Jekyll~ (has dependencies on .gov infrastructure, currently in private beta, possibly only available to  departments and agencies across the US federal government)

Tasks:
- [x] Remove the current Jekyll blog from the existing OperationCode.org Rails app (@rickr should this be a separate GitHub issue ?)
- [x] Set up new Jekyll blog ~at the repo (https://github.com/OperationCode/blog (a Jekyll app that exists separate from existing Rails app)~ at a path within the current Rails app codebase 

~Configure CNAME / DNS settings to redirect https://www.operationcode.org/blog path to the GitHub Pages static site (see https://help.github.com/articles/custom-domain-redirects-for-github-pages-sites/)~
 (not needed if using  `/blog` URL that lives within the main code base)

See #576 (now closed)

**Comment by hollomancer** [comment]
I don't know if Federalist will work with GitHub Pages. If we decide to use it, we can get away with not using Github Pages - may also avoid CNAME/DNS issues?

**Comment by rickr** [comment]
I'm ok removing the old blog with this ticket.

If we want the URL to be /blog the blog will have to live in the main code base somewhere as CNAMEs work on a host level.

**Comment by jjhampton** [comment]
@rickr re: URL at /blog - Yeah, I thought that might be tricky.  Wasn't sure if it was possible.  After reading thru https://help.github.com/articles/custom-domain-redirects-for-github-pages-sites/ with the part that says "Automatically redirected to a subdirectory of an Organization Pages site custom domain (org.example.com/projectname)" I thought it might be possible for it -not- to be in the main code base.

**Comment by jjhampton** [comment]
I'm looking at following this strategy to allow a Jekyll blog to live inside the same codebase as the Rails app, but remain more or less decoupled from the Rails app: 

http://www.nickhammond.com/the-super-simple-way-to-add-a-jekyll-based-blog-in-your-rails-application/

@rickr I'm not very familiar w/ OC's deployment process - when you get a chance, could you let me know if this plan looks feasible? 

TLDR:
* store Jekyll blog in child directory `/blog` of the main app in the GitHub `operationcode` repo  (the blog post above calls for a sibling directory, but I'm thinking this can be done in a child directory)

*  add a deploy script that builds Jekyll and rsyncs it to our local Rails application’s public directory on deploy

* Add the blog/ directory as a linked_directory in the deploy recipe, so we can deploy whenever and the blog directory will persist with the Jekyll blog - (this part I'm unsure of, the blog post above mentions to use this w/ Capistrano, and to my knowledge we're not using that. Is there something similar in our deployment flow that might require this?)

**Comment by rickr** [comment]
Can you configure jekyll to tell it where to send it's built files? I think what we would want to do is have Travis run our jekyll build and store the output in our public dir. We would then configure rails to serve that public directory.

Travis has all sorts of lifecycle hooks we might be able to use: https://docs.travis-ci.com/user/customizing-the-build#The-Build-Lifecycle

**Comment by jjhampton** [comment]
That should work -  there's a `--destination` flag that can be applied to the `jekyll build` command to generate a build into a custom path.  I'll test it out on my dev machine later today.

**Comment by rickr** [comment]
Fixed in #680
